### PR TITLE
XML_Toolkit: Add descriptions

### LINKED ISSUE #306: XML_Toolkit: Change output name for adaptor
Issue raised to change adaptor output name from "X" to "adapter" to match output name with consecutive component.

image

### LINKED ISSUE #307: XML_Toolkit: Adapter inputs
When we change XML adapter name, I do not recall this one?

all my scripts still use old naming and luckily this do not update, as old names make more sense than new ones... we connecting `file name` and not `project name` etc. we need to revert this 

image

### PR DESCRIPTION
**Pull request by LivAndersson**
### Issues addressed by this PR

 Closes #306 
Closes #307 

 
Added descriptions and changed input names for #306 & #307. First BHoM commit yay! 

 ### Test files

Not announced

### CODE REVIEW

File: XML_Adapter/XMLAdapter.cs [hunk 1]
```diff
@@ -31,17 +31,25 @@
...
     {
-        public XMLAdapter(String projectName = "BHoM_gbXML_Export", String SaveDirectoryPath = null, ExportType exportType = ExportType.gbXMLTAS, ExportDetail exportDetail = ExportDetail.Full)
+        [Description("Specify XML file and properties for data transfer")]
+        [Input("xmlFileName", "Name of XML file")]
+        [Input("xmlDirectoryPath", "Path to XML file")]
+        [Input("exportType", "Specify whether the file is TAS or IES specific")]
+        [Input("exportDetail", "Define what you want to include in the export, for example 'Spaces'")]
+        [Output("adapter", "Adapter to XML")]
+
+        public XMLAdapter(String xmlFileName = "BHoM_gbXML_Export", String xmlDirectoryPath = null, ExportType exportType = ExportType.gbXMLTAS, ExportDetail exportDetail = ExportDetail.Full)
```
Comment (FraserGreenroyd) [inline XML_Adapter/XMLAdapter.cs]: Could we remove the `xml` prefix? I think the name of the adapter should be explanatory of the file name and directory path.
Suggest renaming to:
`xmlFileName` -> `fileName`
`xmlDirectoryPath` -> `directoryPath`

File: XML_Adapter/XMLAdapter.cs [hunk 2]
```diff
@@ -31,17 +31,25 @@
...
     public partial class XMLAdapter : BHoMAdapter
     {
-        public XMLAdapter(String projectName = "BHoM_gbXML_Export", String SaveDirectoryPath = null, ExportType exportType = ExportType.gbXMLTAS, ExportDetail exportDetail = ExportDetail.Full)
+        [Description("Specify XML file and properties for data transfer")]
+        [Input("xmlFileName", "Name of XML file")]
+        [Input("xmlDirectoryPath", "Path to XML file")]
+        [Input("exportType", "Specify whether the file is TAS or IES specific")]
+        [Input("exportDetail", "Define what you want to include in the export, for example 'Spaces'")]
+        [Output("adapter", "Adapter to XML")]
+
```
Comment (FraserGreenroyd) [inline XML_Adapter/XMLAdapter.cs]: Can we remove this blank line please :smile:

**Comment by rwemay** [comment]
Welcome and thanks @LivAndersson !! Amazing! 

@FraserGreenroyd thanks for keeping us on the straight and narrow with the whitespace! #spaceterminator

File: XML_Adapter/XMLAdapter.cs (hunk 1 above)
Comment (LivAndersson) [reply XML_Adapter/XMLAdapter.cs]: Not sure I agree here - by specifying xmlFileName we assure that people do not specify the file they want to push/pull to/from XML, but the XML file itself.

File: XML_Adapter/XMLAdapter.cs (hunk 2 above)
Comment (LivAndersson) [reply XML_Adapter/XMLAdapter.cs]: I found a lot of other blank lines in the code. Should I remove them as well while I'm at it? @rwemay  @FraserGreenroyd

Comment (FraserGreenroyd) [reply XML_Adapter/XMLAdapter.cs]: No just this one will be ok, they get caught during these prs and refactoring, so the other lines will be caught by a tidy up later :smile:

File: XML_Adapter/XMLAdapter.cs (hunk 1 above)
Comment (FraserGreenroyd) [reply XML_Adapter/XMLAdapter.cs]: I'm not sure I understand what you mean? If the file is being pulled then it is the file to be pulled from. Similarly, push functionality allows existing files to be added to if they exist. Context from the adapter should confirm which file they're working with in theory?

File: XML_Adapter/XMLAdapter.cs (hunk 2 above)
Comment (LivAndersson) [reply XML_Adapter/XMLAdapter.cs]: Sorted!

**Pull request merged**